src/Symfony/Component/Form/Extension/Validator/EventListener/ValidationListener.php

```diff
@@ -35,17 +41,27 @@ public static function getSubscribedEvents()
         return [FormEvents::POST_SUBMIT => 'validateForm'];
     }
 
-    public function __construct(ValidatorInterface $validator, ViolationMapperInterface $violationMapper)
+    public function __construct(ValidatorInterface $validator, ViolationMapperInterface $violationMapper, ?EventDispatcherInterface $eventDispatcher = null)
     {
         $this->validator = $validator;
         $this->violationMapper = $violationMapper;
+
+        if (!$eventDispatcher) {
+            @trigger_error(sprintf('The "$eventDispatcher" argument to the "%s" constructor will be required in Symfony 6.0.', self::class));
```

Comment:
Now it is recommended to use the `trigger_deprecation` function of the `symfony/deprecation-contracts` package

Reply:
That tells you how old this is and I'm not really staying on top of this. If someone else would like to take it over, please do!